core/src/main/java/io/grpc/internal/ExponentialBackoffPolicy.java

```diff
@@ -39,26 +39,52 @@
 import java.util.concurrent.TimeUnit;
 
 /**
- * Retry Policy for Transport reconnection.  Initial parameters from
- * https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
- *
- * <p>TODO(carl-mastrangelo): add unit tests for this class
+ * Exponential backoff policy for reconnects and call retries.
  */
 final class ExponentialBackoffPolicy implements BackoffPolicy {
-  static final class Provider implements BackoffPolicy.Provider {
+  /**
+   * Provider tuned for connection retries.
+   *
+   * https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+   */
+  static final class ReconnectProvider implements BackoffPolicy.Provider {
+    @Override
+    public BackoffPolicy get() {
+      return new ExponentialBackoffPolicy(new Random(), TimeUnit.SECONDS.toMillis(1),
+              TimeUnit.MINUTES.toMillis(2),  1.6, .2);
+    }
+  }
+
+  /**
+   * Provider tuned for call retries.
+   */
+  static final class CallRetryProvider implements BackoffPolicy.Provider {
     @Override
     public BackoffPolicy get() {
-      return new ExponentialBackoffPolicy();
+      // TODO(lukaszx0) those values are picked arbirarily without much thought behind them. They
+      // need revising after testing call retries.
+      return new ExponentialBackoffPolicy(new Random(), 50, TimeUnit.SECONDS.toMillis(30), 1.2, .2);
```

Comment:
I think we want a multiplier of 2.0. I think 1.2 will backoff too slowly.

Internally, we currently use a jitter of 0.25, but that's probably not worth changing since that's not a value that needs to be nitpicked...